controllers/managedfusionoffering_controller.go

```diff
@@ -180,14 +185,65 @@ func (r *ManagedFusionOfferingReconciler) reconcileSubscription() error {
 	return nil
 }
 
+func (r *ManagedFusionOfferingReconciler) CreateOrUpdate(obj client.Object, f controllerutil.MutateFn) (controllerutil.OperationResult, error) {
+	return ctrl.CreateOrUpdate(r.ctx, r.Client, obj, f)
+}
+
+func (r *ManagedFusionOfferingReconciler) GetAndUpdate(obj client.Object, f controllerutil.MutateFn) error {
+	if f == nil {
+		return fmt.Errorf("MutateFn cannot be nil")
+	}
+	if err := r.get(obj); err != nil {
+		return err
+	}
+	key := client.ObjectKeyFromObject(obj)
+	existing := obj.DeepCopyObject()
+	if err := f(); err != nil {
+		return err
+	}
+	// NamespacedName before and after mutateFunc should remain same
+	if key != client.ObjectKeyFromObject(obj) {
+		return fmt.Errorf("MutateFn cannot mutate object name and/or object namespace")
+	}
+	if equality.Semantic.DeepEqual(existing, obj) {
+		return nil
+	}
+	if err := r.update(obj); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *ManagedFusionOfferingReconciler) list(obj client.ObjectList, listOptions ...client.ListOption) error {
+	return r.Client.List(r.ctx, obj, listOptions...)
+}
```

Comment:
Can we please move these 2 after the simple operations (get, list, update, etc.)?